### PR TITLE
Wait for all routines to finish, don't log errors twice

### PR DESCRIPTION
**Pull request by jharrington22**
This PR fixes an issue were we exit a function closing message channels while go routines using those channels are still running. I've also removed unnecessary logging from those functions since we'll print any errors that arrive on the channel.

### CODE REVIEW

File: pkg/controller/accountclaim/reuse.go [hunk 1]
```diff
@@ -125,7 +125,6 @@ func (r *ReconcileAccountClaim) cleanUpAwsAccount(reqLogger logr.Logger, account
 		case errMsg := <-awsErrors:
 			err = errors.New(errMsg)
 			reqLogger.Error(err, errMsg)
-			return err
```
Comment (rogbas) [inline pkg/controller/accountclaim/reuse.go]: I don't think we can simply remove this return. We still need to inform the caller function that we had an error during the cleanup. That error will trigger the account being flagged as Failed Reuse.

Maybe we could store the error messages and a flag error = true, and after all the 3 functions finish we return it?

Comment (jharrington22) [reply pkg/controller/accountclaim/reuse.go]: Ah I see yes we can just keep a bool indicating there was an error and return once the routines are complete.

Comment (rogbas) [reply pkg/controller/accountclaim/reuse.go]: /lgtm

**Pull request merged**